### ISSUE TITLE
Default scheme colors not used for Social cards when automatic/light dark mode is set

### ISSUE THREAD

**Issue by mitches-got-glitches** [squidfunk/mkdocs-material#6929]
### Context

I'm creating a personal blog with `mkdocs-material==9.5.14` and the built-in Blog plugin.  

### Bug description

If I set my palette as the following with a non-default primary color set in the default scheme, then this color isn't used by the social plugin when creating the social cards, it just uses the default.

```yaml
theme:
  palette:
    - media: "(prefers-color-scheme)"
      toggle:
        icon: material/brightness-auto
        name: Switch to light mode
    - media: "(prefers-color-scheme: light)"
      scheme: default
      primary: black
      accent: orange
      toggle:
        icon: material/brightness-5
        name: Switch to dark mode
    - media: "(prefers-color-scheme: dark)"
      scheme: slate
      toggle:
        icon: material/brightness-4
        name: Switch to system preference
```

![image](https://github.com/squidfunk/mkdocs-material/assets/42358474/a3836656-309d-4cdc-9ed8-c07697f025a7)

If I repeat the palette colors in the top item under palette, then these are used by the social plugin when creating a card.

```yaml
theme:
  palette:
    - media: "(prefers-color-scheme)"
      primary: black
      accent: orange
      toggle:
        icon: material/brightness-auto
        name: Switch to light mode
    - media: "(prefers-color-scheme: light)"
      scheme: default
      primary: black
      accent: orange
      toggle:
        icon: material/brightness-5
        name: Switch to dark mode
    - media: "(prefers-color-scheme: dark)"
      scheme: slate
      toggle:
        icon: material/brightness-4
        name: Switch to system preference
```

![image](https://github.com/squidfunk/mkdocs-material/assets/42358474/d2541492-594f-4769-bbcc-69a8ee249438)

If I move the `media: "(prefers-color-scheme: light)"` item to the top item, then the colors also will be used properly by the social plugin, without the need for repetition.

```yaml
theme:
  palette:
    - media: "(prefers-color-scheme: light)"
      scheme: default
      primary: black
      accent: orange
      toggle:
        icon: material/brightness-5
        name: Switch to dark mode
    - media: "(prefers-color-scheme)"
      toggle:
        icon: material/brightness-auto
        name: Switch to light mode
    - media: "(prefers-color-scheme: dark)"
      scheme: slate
      toggle:
        icon: material/brightness-4
        name: Switch to system preference
```

### Suggested solutions

- When selecting the color scheme to socials, search for the media item with `schema: default` and use those colors. The order of the list in the YAML won't matter with this solution.
- Document the above behaviour in the docs - suggesting either repetition or making sure default scheme is the first item.

### Related links

- [Reporting a bug](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/)
- [Automatic light/dark mode](https://squidfunk.github.io/mkdocs-material/setup/changing-the-colors/?h=dark+mode#automatic-light-dark-mode)

### Reproduction

[9.5.14-media-palette-unused-socials.zip](https://github.com/squidfunk/mkdocs-material/files/14638575/9.5.14-media-palette-unused-socials.zip)

I've put in some commented sections so that anyone checking can try all of the light/dark mode suggestions in the docs and confirm this bug.

### Steps to reproduce

Using the makefile on Linux...

1. `make all` (installs imaging dependencies and requirements into a virtual environment
2. Remove `- info` from the plugins in mkdocs.yml
3. Activate venv with `source .venv/bin/activate`
4. `make docs`

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material/), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/) using the [built-in info plugin](https://squidfunk.github.io/mkdocs-material/plugins/info/).

**Comment by squidfunk** [comment]
Thanks for reporting, and again, thanks for the excellent reproduction and the report + documenting your attempts. Fixed in 66d4500. Now, the social plugin will look for the first palette item that defines a primary color. Note that the social plugin has to employ a heuristic to decide what colors to use, and this might not what is expected. For this reason, the social plugin also allows to explicitly set the [`background_color`](https://squidfunk.github.io/mkdocs-material/plugins/social/#option.background_color) of the social card:

``` yaml
plugins:
  - social:
      cards_layout_options:
        background_color: black
```

If you're using [Insiders](https://squidfunk.github.io/mkdocs-material/insiders/), you can even use CSS colors, including HEX codes or RGB, [as mentioned in our documentation](https://squidfunk.github.io/mkdocs-material/plugins/social/#__code_31_annotation_1):

``` yaml
plugins:
  - social:
      cards_layout_options:
        background_color: "#ffcc00"
```

**Comment by squidfunk** [comment]
[`cae9c18`](https://github.com/squidfunk/mkdocs-material-insiders/commit/cae9c180408967ded8f7560a1ca8134c6e40749d ) also ports the fix to new social plugin implementation in Insiders.

**Comment by squidfunk** [comment]
Released as part of [9.5.15](https://squidfunk.github.io/mkdocs-material/changelog/#9.5.15).
